folium/plugins/time_slider_choropleth.py

```diff
@@ -1,69 +1,30 @@
-        embedded in Leaflet's JavaScript.
-        * If dict, then data will be converted to JSON and embedded
-        in the JavaScript.
-        * If str, then data will be passed to the JavaScript as-is.
-    style_function: function, default None
-        A function mapping a GeoJson Feature to a style dict.
-    name : string, default None
-        The name of the Layer, as it will appear in LayerControls
-    overlay : bool, default False
-        Adds the layer as an optional overlay (True) or the base layer (False).
-    control : bool, default True
-        Whether the Layer will be included in LayerControls
-    smooth_factor: float, default None
-        How much to simplify the polyline on each zoom level. More means
-        better performance and smoother look, and less means more accurate
-        representation. Leaflet defaults to 1.0.
-
-    Examples
-    --------
-    >>> # Providing file that shall be embedded.
-    >>> GeoJson(open('foo.json'))
-    >>> # Providing filename that shall not be embedded.
-    >>> GeoJson('foo.json')
-    >>> # Providing dict.
-    >>> GeoJson(json.load(open('foo.json')))
-    >>> # Providing string.
-    >>> GeoJson(open('foo.json').read())
-
-    >>> # Provide a style_function that color all states green but Alabama.
-    >>> style_function = lambda x: {'fillColor': '#0000ff' if
-    ...                             x['properties']['name']=='Alabama' else
-    ...                             '#00ff00'}
-    >>> GeoJson(geojson, style_function=style_function)
-
-    """
-    def __init__(self, data, styledict, **kwargs):
-        super(TimeDynamicGeoJson, self).__init__(data, **kwargs)
-        assert isinstance(styledict, dict), 'styledict must be a dictionary'
+class TimeSliderChoropleth(GeoJson):
+    def __init__(self, data, styledict, name=None, overlay=True, control=True, **kwargs):
+        super(TimeSliderChoropleth, self).__init__(data, name=name, overlay=overlay, control=control)
+        if not isinstance(styledict, dict):
+            raise ValueError('styledict must be a dictionary, got {!r}'.format(styledict))
         for val in styledict.values():
-            assert isinstance(val, dict), 'each item in styledict must be a dictionary'
+            if not isinstance(val, dict):
+                raise ValueError('Each item in styledict must be a dictionary, got {!r}'.format(val))
+
 
-        self.styledict = styledict
+        # Make set of timestamps.
+        timestamps = set()
+        for feature in styledict.values():
+            timestamps.update(set(feature.keys()))
+        timestamps = sorted(list(timestamps))
 
-        # make set of timestamps
-        self.timestamps = set()
-        for feature in self.styledict.values():
-            self.timestamps.update(set(feature.keys()))
-        self.timestamps = sorted(list(self.timestamps))
+        self.timestamps = json.dumps(timestamps)
+        self.styledict = json.dumps(styledict, sort_keys=True, indent=2)
```

Comment:
By dumping `json` strings we avoid the `u''` in the code and fix the py27 failures.

Reply:
nice